### ISSUE TITLE
Option to disable 

### ISSUE THREAD

**Issue by franktzheng** [typst/typst#3203]
### Description

Would it be possible to add an option to control how `/` is interpreted in a math equation? For instance, it would be nice if something like this could disable automatically interpreting `/` as a fraction:
```typst
#set math.equation(auto-frac: false)
```

### Use Case

There are many cases where it is preferable to display fractions horizontally (i.e., $x / y$) rather than vertically (i.e., $\frac{x}{y}$). In math papers, it is common to typeset inline fractions horizontally, since this keeps the text at a readable size. Moreover, a very important concept in math is that of a [quotient](https://en.wikipedia.org/wiki/Quotient_group), which is almost always denoted as a horizontal fraction rather than a vertical fraction. [Here](https://arxiv.org/pdf/2401.06342.pdf) is a 40-page paper in number theory published today where the symbol `/` appears 202 times, whereas vertical fractions are not used (as far as I know).

While currently it is not difficult to typeset a horizontal fraction by typing `\/`, I believe it would be greatly preferable in many cases if `/` defaulted to displaying a horizontal fraction.

**Comment by franktzheng** [comment]
Perhaps this comment won't be helpful (and whatever I say may be very wrong), but I spent some time digging through the codebase and this is what I gathered. It seems the main difficulty in implementing such an option is that the decision to interpret `/` as a fraction currently occurs at the parser level (see [here](https://github.com/typst/typst/blob/f57c34a7ce7bd09d7c2f2d19f37645ad30493afd/crates/typst-syntax/src/parser.rs#L467)). I guess a quick and dirty fix would be to add an additional `SyntaxKind` called, say, `SlashMathFrac`, and then have this distinction bubble up to wherever the setting `auto-frac` appears. There may be a better way, though.

**Comment by Enter-tainer** [comment]
```
#show math.frac: it => [#it.num #sym.slash #it.denom]

$a+1/(b)$
```
<img width="157" alt="image" src="https://github.com/typst/typst/assets/25521218/fb6f6052-ea02-41ee-af6d-35e845ea9200">

**Comment by Enter-tainer** [comment]
maybe `auto-frac` can add to https://typst.app/docs/reference/math/frac/? like controlling the style of a frac

**Comment by franktzheng** [comment]
> ```
> #show math.frac: it => [#it.num #sym.slash #it.denom]
> 
> $a+1/(b)$
> ```
> 
> <img alt="image" width="157" src="https://private-user-images.githubusercontent.com/25521218/296982095-fb6f6052-ea02-41ee-af6d-35e845ea9200.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTQ1MTEsIm5iZiI6MTcwNTM5NDIxMSwicGF0aCI6Ii8yNTUyMTIxOC8yOTY5ODIwOTUtZmI2ZjYwNTItZWEwMi00MWVlLWFmNmQtMzVlODQ1ZWE5MjAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MzY1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZjkwNWI1MzU0ZjdhNjA3ODdmZmFhODI5OTJjM2Q2ZjQxNDM3YTcxYzVkNzI0MjZiZjY3OTQ3MmFjOGFkNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QNO4sJmERlMoXEir24TAFh7zLl6CcOUcvGgnO7YC0NE">

Thanks! The problem with this would be that it prevents you from displaying vertical fractions at all. I edited my original issue with clarification. I guess if you wanted to still be able to typset fractions, you could just define another `my-frac` function with the definition of the original `frac` function?

**Comment by laurmaedje** [comment]
> Honestly idk if there is a way to temporarily use the default frac variant. Help needed!

I don't think there is except maybe by somehow putting an if into that show rule. Something that distinguishes between the two cases with a label or such.

Other than that, going back to the default rule would need [`revoke` rules](https://github.com/typst/typst/issues/420).

**Comment by jbirnick** [comment]
Just here to support the opinion against automated fractions. In my experience, you rarely need fractions when you write mathematics or computer science papers, but you often need normal slashes for all sorts of quotients like group quotients and so on. So just from the math/cs community side I would expect that they would prefer having no automated fractions.

(Obviously a switchable option would be the best.)

**Comment by cristobaltapia** [comment]
Just my 5 cents. I was having this issue and solved it by simply writing `a\/b`. That shows the normal `/`.

**Comment by jbirnick** [comment]
Sure you can do that, but it would be better if there was an option to disable it. I think what your comment shows is rather that indeed you as an engineer _had_ this issue.

**Comment by cristobaltapia** [comment]
@jbirnick Yes, I'm not arguing that the defaults are good, just that it is easier to go around this with `\/` than the solution proposed above. I think that Typst is trying to be a bit too smart with this type of fractions, as, indeed, they are not generally used in-text.

**Comment by Waschbaar** [comment]
> > ```
> 
> > #show math.frac: it => [#it.num #sym.slash #it.denom]
> 
> > 
> 
> > $a+1/(b)$
> 
> > ```
> 
> > 
> 
> > <img alt="image" width="157" src="https://private-user-images.githubusercontent.com/25521218/296982095-fb6f6052-ea02-41ee-af6d-35e845ea9200.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTQ1MTEsIm5iZiI6MTcwNTM5NDIxMSwicGF0aCI6Ii8yNTUyMTIxOC8yOTY5ODIwOTUtZmI2ZjYwNTItZWEwMi00MWVlLWFmNmQtMzVlODQ1ZWE5MjAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MzY1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZjkwNWI1MzU0ZjdhNjA3ODdmZmFhODI5OTJjM2Q2ZjQxNDM3YTcxYzVkNzI0MjZiZjY3OTQ3MmFjOGFkNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QNO4sJmERlMoXEir24TAFh7zLl6CcOUcvGgnO7YC0NE">
> 
> 
> 
> Thanks! The problem with this would be that it prevents you from displaying vertical fractions at all. I guess if you wanted to still be able to typeset fractions, you could just define another `my-frac` function with the definition of the original `frac` function?

Another problem with this workaround is that the parentheses get automatically removed. A lot of times I want to have parentheses in horizontal fractions when I write complicated formulas involving the quotient operation. It would be very counterintuitive if I have to use double parentheses in these cases.

There is no easy fix in my head. I don't need the vertical fractions at all. So I can use an auto snippet to escape all occurrences of the slash that follows a white space. This is, of course, far from ideal.

I think the horizontal fraction appears so common in modern mathematical literature that it deserves proper support from Typst. I can imagine that the issue might go as deep as to the parser level, but please fix it. The best solution is probably to add an option to the compiler that disables (or rather, enables, to be more realistic) the syntax with / altogether.

**Comment by jbirnick** [comment]
> It could be useful to have `/` represent just a slash, while `//` invokes `frac`

Yeah would be cool but unfortunately `//` already is reserved for comments.

**Comment by Waschbaar** [comment]
> > It could be useful to have `/` represent just a slash, while `//` invokes `frac`
> 
> 
> 
> Yeah would be cool but unfortunately `//` already is reserved for comments.

I prefer `/` for a slash and `\/` for invoking frac, the opposite of what we have now.

**Comment by camelid** [comment]
> > It could be useful to have `/` represent just a slash, while `//` invokes `frac`
> 
> Yeah would be cool but unfortunately `//` already is reserved for comments.

Ah, right. Still, perhaps something similar like @Waschbaar's suggestion could be good.

**Comment by mkorje** [comment]
> Another problem with this workaround is that the parentheses get automatically removed. A lot of times I want to have parentheses in horizontal fractions when I write complicated formulas involving the quotient operation. It would be very counterintuitive if I have to use double parentheses in these cases.

I've thought a bit about this issue but I am leaning towards it not being a problem. With inline fractions, users should be forced to properly delimit what is part of the fraction. As with all the other syntax in math, the first set of parentheses are used for grouping, not displaying parentheses. Thus I think we should maintain this consistency, especially as code for a fraction should be consistent regardless of whether it is inline or display, in my opinion. I think having to do the 'counterintuitive' double parentheses in these cases is a necessary evil then, as is needed everywhere else in math.

Of course if a user wants a literal backlash, they can escape it. But this should not be encouraged as it is 'lossy'; the compiler doesn't have any information about the fraction then.

I think more fine grained control based on whether one is in inline or display math should be possible if these (currently hidden) fields are moved to a separate element, though I'll have to investigate some more to see if this will work as wanted.

**Comment by jbirnick** [comment]
Let's say I want to write inline "fractions", so that it's just rendered with a backslash, and I only want to use the `/` symbol (without escaping it). So I want to write `3/4` and it should be rendered as 3/4, with a backslash.

Then why does the compiler need to know that it's a fraction? Would you render the thing described above differently than `3 \/ 4`?

It seems like that you want that "disabling automatic fractions" has the impact to still store things as `FracElem`, but just with some field `flat: true` or so. But I would argue that this is not what I want. My understanding would be that "disabling automatic fractions" means that `/` is not even creating a `FracElem` anymore, instead it should just literally render a backslash.

I want to disable the smartness completely, not have _another_ smartness. I just want to render backslashes as backslashes..

**Comment by jbirnick** [comment]
I think the best solution would be have different syntax for automatic fractions. So that `/` just renders a slash and we have a special syntax for automatic fractions.

- A user previously suggested `//`, which is amazing, except that it's unfortunately already used for comments. :/
- As a programmer I would be happy to do something like `/a` for automatic fractions (e.g. `3 /a 4`). But I'm not sure if it's the best for non-technical users. Other choices are `/:` or `/@`.
- Or something like `[/]`, though it's 3 symbols so not so easy to type.

**Comment by laurmaedje** [comment]
> Thus I think we should maintain this consistency, especially as code for a fraction should be consistent regardless of whether it is inline or display, in my opinion.

Isn't that precisely the problem though? The fraction itself should be written in the same way, just its display representation would be different depending on the style. If you need to adjust the number of parentheses depending on the style that doesn't work anymore. I think it would make more sense to keep exactly the parentheses the user specified in inline style but trim them in display style.

**Comment by Waschbaar** [comment]
> > Another problem with this workaround is that the parentheses get automatically removed. A lot of times I want to have parentheses in horizontal fractions when I write complicated formulas involving the quotient operation. It would be very counterintuitive if I have to use double parentheses in these cases.
> 
> I've thought a bit about this issue but I am leaning towards it not being a problem. With inline fractions, users should be forced to properly delimit what is part of the fraction. As with all the other syntax in math, the first set of parentheses are used for grouping, not displaying parentheses. Thus I think we should maintain this consistency, especially as code for a fraction should be consistent regardless of whether it is inline or display, in my opinion. I think having to do the 'counterintuitive' double parentheses in these cases is a necessary evil then, as is needed everywhere else in math.

I'm talking about the horizontal fraction (i.e. a literal /) and the vertical fraction (i.e. the latex `\frac`).
The problem is that the horizontal fraction does not visually show the grouping without parenthesis. The vertical one does. They are different and I doubt if it's a good idea to treat them in the same way. The "necessary evil" is likely going to make Typst very annoying to use for the algebraic community.

> Of course if a user wants a literal backlash, they can escape it. But this should not be encouraged as it is 'lossy'; the compiler doesn't have any information about the fraction then.

Typst, as a typesetting tool, does not need to (and of course will never) understand the proofs I write. There is a boundary which separates the information that is important to Typst from that is not. If some information is never going to make a difference in the visuals, then it can be safely ignored by Typst. I think whether a literal / is a fraction or not falls into this category.

I think making a keyword `frac` and use it in the same way as the `/` today is a good idea. It makes sense that if you want a more sophisticated visual effect, then you need to put in more code.

**Comment by laurmaedje** [comment]
> If some information is never going to make a difference in the visuals, then it can be safely ignored by Typst. I think whether a literal / is a fraction or not falls into this category.

It is important to make the PDF output accessible (e.g. for people that use a screen reader) and for HTML export.

**Comment by Waschbaar** [comment]
> > If some information is never going to make a difference in the visuals, then it can be safely ignored by Typst. I think whether a literal / is a fraction or not falls into this category.
> 
> It is important to make the PDF output accessible (e.g. for people that use a screen reader) and for HTML export.

I don't think a horizontal fraction, denoting the quotient operation, would ever be rendered as anything other than the literal /. It is a literal / in PDF documents, a literal / on a screen reader and still a literal / in HTML.

**Comment by laurmaedje** [comment]
Perhaps that's true for `/` (not sure how MathML treats inline fractions), but I just wanted to bring up that this sentence is not generally true:

> If some information is never going to make a difference in the visuals, then it can be safely ignored by Typst.

**Comment by mkorje** [comment]
> Then why does the compiler need to know that it's a fraction? Would you render the thing described above differently than `3 \/ 4`?

I think @laurmaedje has better said what I was really only implying in my response; the output Typst produces should be accessible!

I think my view is best described by way of an example: consider `$ I_K (f) sect P_K / P_(K, ZZ) (f) $` $$I_K (f) \cap P_K / P_{K, \mathbb{Z}} (f)$$ (this is real notation!). If Typst does not create a `FracElem` at all (i.e. the use of a literal slash `\/` currently), there is no semantic information at all (and so bad for accessibility). Thus I don't think think this is good behavior and it should not be encouraged in Typst; hence I think inline fractions should be an element, a `FracElem`. Following LaTeX and saying a literal slash is the right way to do inline fractions is not the way to go imo.

Ok, so if we say inline fractions are elements then I'll want to group with parentheses what should be considered the numerator and denominator of the fraction: `$ (I_K (f) sect P_K) / (P_(K, ZZ) (f)) $`. I've now made it clear what is part of the fraction and this is great for accessibility! But I _don't_ want these parentheses to be displayed $$\left( I_K (f) \cap P_K \right) / \left( P_{K, \mathbb{Z}} (f) \right)$$ (I think this is quite ugly as the parentheses make things too cluttered. In fact, the book where this notation is from typesets this without the parentheses.).

> Isn't that precisely the problem though? The fraction itself should be written in the same way, just its display representation would be different depending on the style. If you need to adjust the number of parentheses depending on the style that doesn't work anymore. I think it would make more sense to keep exactly the parentheses the user specified in inline style but trim them in display style.

Just like with everything else in math, parentheses are used to group things first and doubling up is a necessary evil if I want them to be displayed (e.g. `A_((x + y))` and `R / ((I))` - this is a display fraction). While I get that for inline fractions it is slightly different - the parentheses being there is expected so that it is clear what is part of the fraction - but I don't think this is universal (I've shown at least one example above). This, paired with keeping the behavior of parentheses consistent with the rest of the math, in my opinion outweighs not needing to modify the fraction when changing the style. After all the behavior is still consistent, despite maybe being unexpected as it does not match its display representation.

**Comment by Waschbaar** [comment]
> > Then why does the compiler need to know that it's a fraction? Would you render the thing described above differently than `3 \/ 4`?
> 
> I think [@laurmaedje](https://github.com/laurmaedje) has better said what I was really only implying in my response; the output Typst produces should be accessible!
> 
> I think my view is best described by way of an example: consider `$ I_K (f) sect P_K / P_(K, ZZ) (f) $` I K ( f ) ∩ P K / P K , Z ( f ) (this is real notation!). If Typst does not create a `FracElem` at all (i.e. the use of a literal slash `\/` currently), there is no semantic information at all (and so bad for accessibility). Thus I don't think think this is good behavior and it should not be encouraged in Typst; hence I think inline fractions should be an element, a `FracElem`. Following LaTeX and saying a literal slash is the right way to do inline fractions is not the way to go imo.

I don't get why skipping the `FracElem` is bad for accessibility. Can you give one concrete example (a specific reading device?) where people will be annoyed by not having this semantic information? After all I typeset my documents for other human beings, not computers, to read. And there is just no way a simple language like Typst can capture all the structures in the expressions I write.

As for the example, I think it is more about another questionable design decision of Typst: using the same parenthesis `()` for both grouping the expression for the compiler and displaying an actual parenthesis. Sometimes the parenthesis appears and sometimes it does not. Adding something near the parenthesis can affect its behavior. I know that there is a specific rule determining when it appears and when it does not. And I know that typing double parenthesis can force it to appear. However research is difficult and I want to treat myself nicely when it comes to writing. Oh the double parenthesis is a mathematical operation in itself. Latex uses a less frequently appearing parenthesis `{}` for grouping, and makes an easy and consistent convention that a literal `{}` is displayed by escaping.

The Typst developers are free to say that having to use double parenthesis is a necessary evil. But it's annoying enough that the majority of algebraists will not even consider giving Typst a try just because of this evil.

**Comment by jbirnick** [comment]
> I think my view is best described by way of an example: consider `$ I_K (f) sect P_K / P_(K, ZZ) (f) $`

Wait now I'm completely confused. Because this is an algebraic example, where it denotes a quotient rather than a fraction. My understanding would be that you want to use `\/` here because it doesn't denote a fraction, so it should not semantically be captured by `FracElem`, right?

Also then you would completely lose the ability to render _actual_ nice fractions, no? Because I guess your group quotient `$ I_K (f) sect P_K / P_(K, ZZ) (f) $` should be rendered in display mode still with a _slash_, and not as a big fraction with a line in the middle. (Because that's how you always render group quotients.) So you would make fractions render flat _everywhere_. But then you lose the ability to actually typeset a nice fraction in display mode, no?

> It is important to make the PDF output accessible (e.g. for people that use a screen reader) and for HTML export.

That's understandable on the one hand, but on the other hand:
- There are many many other examples where Typst won't capture the semantics. And it's impossible to always do. E.g. consider `integral ... dif x`. Then the `dif x` belongs to the `integral`, but this is not "stored" anywhere in Typst. When you do multiple integrals like `integral ... integral dif x dif y` it's not clear which `dif` belongs to which `integral`, and indeed people have different conventions, and it's impossible to capture this in Typst. And so many other examples. I don't know if having semantics for inline fractions instead of not having them doesn't make a big difference.
- Why would that actually be necessary for screen readers in this case? I think here I agree with @Waschbaar that all semantics are already captures in visuals (i.e. rendered symbols). When there is an equation like `f = x z / y` then why should there be any grouping under the hood? If a screen reader just reads symbol after symbol out loud, it's clear what is meant. Conversely, I claim that if reading out loud leaves ambiguity, then there is also ambiguity in the visuals for people who can see the equation. Or do you have a counterexample?

I second again that algebraic people 99% of the time want to render quotients of algebraic objects, not fractions, so they would have to type `\/` all the time. It would be great to be able to just write `/`, and have it interpreted as a _slash_, and not as a fraction at all.

**Comment by Waschbaar** [comment]
> > I think my view is best described by way of an example: consider `$ I_K (f) sect P_K / P_(K, ZZ) (f) $`
> 
> Wait now I'm completely confused. Because this is an algebraic example, where it denotes a quotient rather than a fraction. My understanding would be that you want to use `\/` here because it doesn't denote a fraction, so it should not semantically be captured by `FracElem`, right?
> 
> Also then you would completely lose the ability to render _actual_ nice fractions, no? Because I guess your group quotient `$ I_K (f) sect P_K / P_(K, ZZ) (f) $` should be rendered in display mode still with a _slash_, and not as a big fraction with a line in the middle. (Because that's how you always render group quotients.) So you would make fractions render flat _everywhere_. But then you lose the ability to actually typeset a nice fraction in display mode, no?

I think that example is to demonstrate that there are cases where @mkorje believes that the grouping information should be retained by the compiler while the parenthesis is usually omitted in the print. In that example the current rule of omitting parentheses happens to produce the correct result. It happens to. When it doesn't, the user would have to fix it.



> * Why would that actually be necessary for screen readers in this case? I think here I agree with [@Waschbaar](https://github.com/Waschbaar) that all semantics are already captured in visuals (i.e. rendered symbols). When there is an equation like `f = x z / y` then why should there be any grouping under the hood? If a screen reader just reads symbol after symbol out loud, it's clear what is meant. Conversely, I claim that if reading out loud leaves ambiguity, then there is also ambiguity in the visuals for people who can see the equation. Or do you have a counterexample?

Also I don't think a screen reader should read out a parenthesis that is not in the print in this case. Just treat the slash the same way as any other operator that Typst has no native support for. Any example where an expression involving a slash is unambiguous in the print but ambiguous in the screen reading can be adapted to be such an example involving the tensor product, the direct sum, the arrows, the limits notations etc. Shall we teach Typst about all of these?

**Comment by xyguo** [comment]
Just curious, what's the current best way to input a normal slash? Asking because I'm typing something involves lots of quotient rings $\mathbb{Z}[x] / f$ and it definitely shouldn't appear like a vertical `\frac`. Right now I just escape the `/` as `\/`, but wondering if there's a better way.

**Comment by bluss** [comment]
@xyguo Consider these alternatives and whether you think they are better than what you have:

```typst
$ZZ[x] slash f$

#let quot(a, b) = $#a slash #b$
$quot(ZZ[x], f)$
```

**Comment by cAttte** [comment]
here is a way to easily use both notations in a document, following this idea from above:

> Something that distinguishes between the two cases with a label or such.

```typ
#let frac(..args) = [#math.frac(..args)<explicit-frac>]
#show math.frac: it => {
  if it.has("label") and it.label == <explicit-frac> {
    it
  } else {
    it.num + sym.slash + it.denom
  }
}

$ a/b != frac(a, b) $
```

<img width="1587" height="240" alt="Image" src="https://github.com/user-attachments/assets/a33b5859-9713-4375-80bd-5c8745a8841d" />

though it does require double-parenthesizing, which sucks

**Comment by mkorje** [comment]
I tried to investigate some more about screen readers and MathML output to give some relevant examples in the wild.

<details>

<summary>

An example from [arXiv:2507.23046](https://arxiv.org/abs/2507.23046)

</summary>

Consider the following piece of math:

<img width="2816" height="300" alt="Image" src="https://github.com/user-attachments/assets/85ac3211-29a2-45be-a096-0e1f01c29abb" />

Translation into Typst currently:
```typ
$ abs(f(x) - f(y)) >= abs(f(x) - f(y)) \/ d_H(x, y) >= (1 \/ n) abs(f(x) - f(y)), $
```
And in the TeX source:
```tex
 \[
  |f(x) - f(y)| \geq |f(x) - f(y)|/d_H(x,y)  \geq (1/n) |f(x) - f(y)|   ,
 \]
```

The MathML arXiv currently produces isn't great. The parsing can't tell that `d_H(x,y)` is one whole expression so it gets split. Following [3.3.1.3 Proper grouping of sub-expressions using `<mrow>`](https://w3c.github.io/mathml/#presm_grouping_sub_expressions) it appears then that the denominator of the fraction is incorrectly taken to be only `d_H`.

```html
<mrow>
    <mrow>
        <mrow>
          ...
        </mrow>
        <mo mathsize="1.200em">≥</mo>
        <mrow>
            <mrow>
                <mrow>
                  ...
                </mrow>
                <mo maxsize="1.200em" minsize="1.200em" stretchy="true" symmetric="true">/</mo>
                <msub>
                    <mi mathsize="1.200em">d</mi>
                    <mi mathsize="1.200em">H</mi>
                </msub>
            </mrow>
            <mo lspace="0em" rspace="0em">​</mo>
            <mrow>
                <mo maxsize="1.200em" minsize="1.200em">(</mo>
                <mi mathsize="1.200em">x</mi>
                <mo mathsize="1.200em">,</mo>
                <mi mathsize="1.200em">y</mi>
                <mo maxsize="1.200em" minsize="1.200em">)</mo>
            </mrow>
        </mrow>
        <mo mathsize="1.200em">≥</mo>
        <mrow>
            <mrow>
              ...
            </mrow>
            <mo lspace="0em" rspace="0em">​</mo>
            <mrow>
              ...
            </mrow>
        </mrow>
    </mrow>
    <mo mathsize="1.200em">,</mo>
</mrow>
```

However, the speech MathCAT (https://nsoiffer.github.io/MathCATDemo/) yields from this MathML still appears fine (maybe though the "divide by ..." not capturing the `(x, y)` is a problem?)
> the absolute value of f x minus f y, end absolute value; is greater than or equal to; the absolute value of f x minus f y, end absolute value; divided by d sub  cap h; open paren x comma, y, close paren; is greater than or equal to; open paren 1 divided by n, close paren; the absolute value of f x minus f y, end absolute value

</details>

<details>

<summary>

An example from [arXiv:2506.14139](https://arxiv.org/abs/2506.14139)

</summary>

Consider the following piece of math:

<img width="1878" height="600" alt="Image" src="https://github.com/user-attachments/assets/0ee1210e-a9be-4be5-86aa-39cee067cee6" />

Translation into Typst currently:
```typ
#let SL = math.op("SL")
$ SL_2(ZZ) \/ plus.minus Gamma_1(N) $
```
And in the TeX source:
```tex
$\mathrm{SL}_2(\mathbb{Z})/\pm\Gamma_1(N)$
```

Surprisingly (?), the MathML arXiv produces is "flat".

```html
<mrow id="S7.Thmtheorem2.p1.1.m1.1b">
    <msub id="S7.Thmtheorem2.p1.1.m1.1.2">
        <mi id="S7.Thmtheorem2.p1.1.m1.1.2.2">SL</mi>
        <mn id="S7.Thmtheorem2.p1.1.m1.1.2.3">2</mn>
    </msub>
    <mrow id="S7.Thmtheorem2.p1.1.m1.1.3">
        <mo id="S7.Thmtheorem2.p1.1.m1.1.3.1" stretchy="false">(</mo>
        <mi id="S7.Thmtheorem2.p1.1.m1.1.1">ℤ</mi>
        <mo id="S7.Thmtheorem2.p1.1.m1.1.3.2" stretchy="false">)</mo>
    </mrow>
    <mo id="S7.Thmtheorem2.p1.1.m1.1.4" rspace="0em">/</mo>
    <mo id="S7.Thmtheorem2.p1.1.m1.1.5" lspace="0em">±</mo>
    <msub id="S7.Thmtheorem2.p1.1.m1.1.6">
        <mi id="S7.Thmtheorem2.p1.1.m1.1.6.2" mathvariant="normal">Γ</mi>
        <mn id="S7.Thmtheorem2.p1.1.m1.1.6.3">1</mn>
    </msub>
    <mrow id="S7.Thmtheorem2.p1.1.m1.1.7">
        <mo id="S7.Thmtheorem2.p1.1.m1.1.7.1" stretchy="false">(</mo>
        <mi id="S7.Thmtheorem2.p1.1.m1.1.7.2">N</mi>
        <mo id="S7.Thmtheorem2.p1.1.m1.1.7.3" stretchy="false">)</mo>
    </mrow>
</mrow>
```

And the speech MathCAT gives from this MathML looks fine:
> SL sub 2, of  cap z, divided by, plus or minus,  cap gamma sub 1, of  cap n

</details>

TLDR: Re-adding the parentheses seems fine, even from an accessibility perspective. While it is a bit inconsistent I guess that something like `$ f / g_1(x) $` won't display "consistently" between styles, if we instead absorbed the parentheses we just end up with the same issue the other way around in something like `$ (a + b) / $`.
